cmd/ctl/check_config.go

```diff
@@ -0,0 +1,580 @@
+			if strings.TrimSpace(globalOpts.agentContext) == "" ||
+				strings.TrimSpace(globalOpts.agentNamespace) == "" ||
+				strings.TrimSpace(globalOpts.principalContext) == "" ||
+				strings.TrimSpace(globalOpts.principalNamespace) == "" {
+				cmdutil.Fatal("--agent-context, --agent-namespace, --principal-context, --principal-namespace are all required")
+			}
+			ctx := context.TODO()
+			agentClt, err := kube.NewKubernetesClientFromConfig(ctx, globalOpts.agentNamespace, "", globalOpts.agentContext)
+			if err != nil {
+				cmdutil.Fatal("Could not create agent Kubernetes client: %v", err)
+			}
+			principalClt, err := kube.NewKubernetesClientFromConfig(ctx, globalOpts.principalNamespace, "", globalOpts.principalContext)
+			if err != nil {
+				cmdutil.Fatal("Could not create principal Kubernetes client: %v", err)
+			}
+			// Run principal checks as part of agent checks
+			results := []checkResult{}
+			results = append(results, RunPrincipalChecks(ctx, principalClt, globalOpts.principalNamespace)...)
+			results = append(results, RunAgentChecks(ctx, agentClt, globalOpts.agentNamespace, principalClt, globalOpts.principalNamespace)...)
+			printResultsAndExit(results)
+		},
+	}
+	return command
+}
+
+func printResultsAndExit(results []checkResult) {
+	hasErr := false
+	fmt.Println("Configuration validation results:")
+	for _, r := range results {
+		fmt.Println(r.String())
+		if r.err != nil {
+			hasErr = true
+		}
+	}
+	if hasErr {
+		cmdutil.Fatal("one or more checks failed")
+	}
+}
+
+// getArgoCDCR retrieves the ArgoCD CR from the specified namespace, if it exists.
+func getArgoCDCR(ctx context.Context, kc *kube.KubernetesClient, ns string) (*unstructured.Unstructured, error) {
+	if kc.DynamicClient == nil {
+		return nil, fmt.Errorf("dynamic client is not available")
+	}
+	gvr := schema.GroupVersionResource{Group: "argoproj.io", Version: "v1beta1", Resource: "argocds"}
+	list, err := kc.DynamicClient.Resource(gvr).Namespace(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		// If CR doesn't exist, return nil, nil to indicate "not found, use defaults"
+		if k8serrors.IsNotFound(err) {
+			return nil, nil
+		}
+		// For other errors, return the error
+		return nil, fmt.Errorf("failed to list ArgoCD CRs: %w", err)
+	}
+	if list == nil || len(list.Items) == 0 {
+		return nil, nil // CR not found, use defaults
+	}
+	// Use the ArgoCD CR found in the namespace
+	return &list.Items[0], nil
+}
+
+// getPrincipalSecretNames reads secret names from ArgoCD CR if available, otherwise returns defaults.
+func getPrincipalSecretNames(ctx context.Context, kc *kube.KubernetesClient, ns string) (string, string, string, string) {
+	caSecretName := config.SecretNamePrincipalCA
+	tlsSecretName := config.SecretNamePrincipalTLS
+	proxyTLSSecretName := config.SecretNameProxyTLS
+	jwtSecretName := config.SecretNameJWT
+
+	// Try to read from ArgoCD CR
+	cr, err := getArgoCDCR(ctx, kc, ns)
+	if err != nil {
```

Comment:
We should return an error here if err is non-nil.

Or, said another way:

We shouldn't assume that an error means the ArgoCD CR isn't available. It may just be that another error occurred.

If you want to know whether ArgoCD CRs could not be found, for example, you can instead return a boolean from `getArgoCDCR`